### PR TITLE
Fix collectibles autodetection

### LINKED ISSUE #3335: NFT Caching issue
### **Description**

- NFTs appearing into another accounts and networks, this happens because of an edge async issue between the CollectibleDetectionController, CollectiblesController, and current address and network. This can be verified by changing account/network right after starting the auto detection mechanism by refreshing the wallet view.
- This development also addresses the following breaking changes from the controllers version 21.0.1
  - https://github.com/MetaMask/controllers/pull/645
  - https://github.com/MetaMask/controllers/pull/636
  - https://github.com/MetaMask/controllers/pull/638
  - https://github.com/MetaMask/controllers/pull/655

### **Technical Details**

- Add a check to verify that the NFT is being added in the correct account and network that started the request (controllers layer).
- Add unit and integration testing to the new development and/or update existing tests.

### **Acceptance Criteria**

- NFTs should show up on the correct accounts and networks

### **References**

- [Dave Pazdan's Issue #3033](https://github.com/MetaMask/metamask-mobile/issues/3033)
- [Dan Finlay's Issue #1663](https://github.com/MetaMask/metamask-mobile/issues/1663)
- [Issue #3406](https://github.com/MetaMask/metamask-mobile/issues/3406)

### PR DESCRIPTION
**Pull request by wachunei**
**Description**

This PR includes controllers from https://github.com/MetaMask/controllers/pull/636 and adds selectors 

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #3335 

**To do**
- [x] Remove controllers from package json once they are released

### CODE REVIEW

File: app/core/Engine.js [hunk 1]
```diff
@@ -69,6 +69,8 @@ class Engine {
 					useStaticTokenList:
 						initialState?.PreferencesController?.useStaticTokenList === undefined ||
 						initialState.PreferencesController.useStaticTokenList,
+					// TODO: Use previous value when preferences UI is available
+					useCollectibleDetection: true,
```
Comment (gantunesr) [inline app/core/Engine.js:73]: Note:  This development should be addressed in #2733

**Review by gantunesr — COMMENTED** [review]
LGTM. I'll approve it when this is no longer a draft.

**Comment by wachunei** [comment]
Reminder we need to remove controllers from package json once they are released

File: app/core/Engine.js (hunk 1 above)
Comment (wachunei) [reply app/core/Engine.js:73]: TODO: `openSeaEnabled: true` must be added too

**Review by gantunesr — APPROVED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]
🌮 🌮 🌮 🌮

**Pull request merged**